### DISCUSSION TITLE
Kernel panic on boot to hard drive with last ipxe version

### DISCUSSION THREAD

**Discussion by Tony-Jab** [ipxe/ipxe#1057] [General]
Hi,
I have a problem with the last ipxe version.
When i try to boot on the disk in my menu, i get an kernel panic after grub choice (ubuntu 22.04 LTS) : 
Kernel panic - not syncing: VFS: Unable to mount root fs or unkown-block(0,0)
With an older ipxe version, i can boot to my device (but i can't keep them because i have a problem for to boot in pxe on some model Dell and HP..).
Have you had similar problems?
Just before having the kernel panic, I have 2 lines displayed that I did not have with the old ipxe version : 
EFI stub: Loaded initrd from LINUX_EFI_INITRD_MEDIA_GUID device path
EFI stub: Measured initrd data into PCR 9

Thanks alot !

**Comment by NiKiZe** [comment]
You can find other posts about LINUX_EFI_INITRD_MEDIA_GUID by using the search function 

But could you post which kernel version you are using as well as the script you are using at least relevant kernel and initrd lines and any related lines.

**Reply by Tony-Jab** [reply]
Thanks for your answer,
I have search other posts about "LINUX_EFI_INITRD_MEDIA_GUID" but i didn't found a solution.
i try this kernel : initrd.img-6.0.0-1016-oem, initrd.img-6.2.0-34-generic and initrd.img-5.15
pxe line : 
```
echo "choose --default hddboot --timeout 10000 target && goto \${target}\n";
echo ":hddboot\n";
echo "iseq \${platform} pcbios && sanboot --no-describe --drive 0x80 || imgfetch refind.conf && chain -ar refind_x64.efi \n";
```

As I said, it is to boot from the disk via pxe (in this case, if after 10 seconds there was no choice in the menu).
If I boot directly from the disk when starting the pc it works

**Reply by NiKiZe** [reply]
Ok, so you do a imgfetch and then chain refind, not sure what that is, but add an imgstat before the chain.
My apologies for missing the "disk" part

Background, in EFI mode you have been required to specify the filename of any initrd on the commandline so the kernel knows what to look for. LINUX_EFI_INITRD_MEDIA_GUID provides a different method to supply this.

iPXE might provide this protocol for any loaded initrd (imgfetch is a alias) and so your kernel loads this.
Check the bootloader you are using to make sure it gives the correct initrd, might need to investigate further how to disable MEDIA_GUID

**Reply by Tony-Jab** [reply]
Good morning,
To continue our discussion, I have been using this method for 5 years without problem.
The problem comes from the latest version of pxe.
If I use my old version of 2021 I can boot on computers on ubuntu 22.04LTS.
If I use the latest version I have a kernel panic. The problem is that I have to update it to be able to boot in pxe on the latest machines..
I saw this thread (https://github.com/ipxe/ipxe/commit/09e8a1540) but I didn't understand everything.
I tried looking on the OS side (kernel, grub, etc.) but the problem does not come from there.

**Reply by NiKiZe** [reply]
Yes, this is a new feature, and you are correct in that the behaviour you are seeing is an unwanted and less considered side effect of the feature, we do not consider this at the level of bug.

The path of disk booting from iPXE are not really "supported". And that is probably also why you are using refind ;)

Maybe if you update us with the full chain of boot events you are using, maybe we can find a solution. And again output of ifstat, would be appreciated, maybe there is a easy explanation and fix there.

**Reply by Tony-Jab** [reply]
You would like the result of ifstat pxe command?
I may not have understood well, if you could tell me in detail what information you want, I can give it to you.

**Reply by NiKiZe** [reply]
Before your chain command, add ifstat, record the output.

**Reply by Tony-Jab** [reply]
![image](https://github.com/ipxe/ipxe/assets/91202256/24b5e840-e5e2-427c-aeb1-2eee2d71d017)

**Reply by NiKiZe** [reply]
I'm terribly sorry, I mixed this up.
imgstat is the one we need to see.

**Reply by Tony-Jab** [reply]
![image](https://github.com/ipxe/ipxe/assets/91202256/805357bd-d760-45ee-ae78-a6b3527f5958)

**Reply by Tony-Jab** [reply]
Good morning,
Do you have any news regarding my problem?
Thanks alot
